tests/js/test-utils.js

```diff
@@ -18,19 +18,43 @@
 
 /* eslint-env es6, node */
 
-'use strict';
+"use strict";
 
-module.exports = {
-    uuid: function() {
-        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
-            let r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
-            return v.toString(16);
-        });
-    },
-    genPartition: function() {
-        return 'xxxxxxxxxxxx4xxxyxxxxxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
-            let r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
-            return v.toString(16);
-        });
+function uuid() {
+    return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function(c) {
+        let r = Math.random() * 16 | 0, v = c == "x" ? r : (r & 0x3 | 0x8);
+        return v.toString(16);
+    });
+}
+
+function genPartition() {
+    return "xxxxxxxxxxxx4xxxyxxxxxxxxxxxxxxx".replace(/[xy]/g, function(c) {
+        let r = Math.random() * 16 | 0, v = c == "x" ? r : (r & 0x3 | 0x8);
+        return v.toString(16);
+    });
+}
+
+function randomVerifiableEmail() {
+    // according to the custom register function, emails will register if they contain "realm_tests_do_autoverify"
+    return `realm_tests_do_autoverify_${uuid()}_@test.com`;
+}
+
+async function getRegisteredEmailPassCredentials(app) {
+    if (!app) {
+      throw new Error("No app supplied to 'getRegisteredEmailPassCredentials'");
```

Comment:
maybe "Undefined argument app"

Reply:
considering this is for testing it does not matter much

Reply:
I was trying to make the error message as explicit as possible - it's only in the case _we_ accidentally call `Utils. getRegisteredEmailPassCredentials()` (without args).
Maybe we should introduce a `TestSetupError`, and throw an error of that type in these cases? (Would be quite easy to spot when writing tests).